src/scripts/gym/Gym.ts

```diff
@@ -14,11 +15,46 @@ interface gymFlags {
 /**
  * Gym class.
  */
-class Gym {
+class Gym extends TownContent {
+    buttonText: string;
+    public tooltip = 'Battle Gym Leaders to earn badges';
+    public cssClass() {
+        return ko.pureComputed(() => {
+            if (App.game.badgeCase.hasBadge(this.badgeReward)) {
+                return 'btn btn-success';
+            }
+            return 'btn btn-secondary';
+        });
+    }
+    public text(): string {
+        return this.buttonText;
+    }
+    public isVisible(): boolean {
+        return true;
+    }
+    public onclick(): void {
+        GymRunner.startGym(this);
+    }
     public flags = {
         quest: true,
         achievement: true,
     };
+    public showAutoStart() {
+        return App.game.statistics.gymsDefeated[GameConstants.getGymIndex(this.town)]() >= 100;
+    }
+    public onAutoStart() {
+        GymRunner.startGym(this, true);
+    }
+    public areaStatus(): areaStatus {
+        if (Gym.isUnlocked(this)) {
+            if (!App.game.badgeCase.hasBadge(this.badgeReward)) {
+                return areaStatus.unlockedUnfinished;
+            } else if (!Gym.isAchievementsComplete(this)) {
+                return areaStatus.missingAchievement;
+            }
+        }
+        return areaStatus.completed;
```

Comment:
Should this be returning `locked` if it's not unlocked yet?

Reply:
If the town is unlocked, but the gym is not, i don't want the town to be greyed out, like you can't access it. That was atleast the idea